src/routes/api.ts

```diff
@@ -0,0 +1,37 @@
+import { FastifyInstance, FastifyReply, FastifyRequest } from 'fastify';
+import ConfigFactory from '../config/config-factory';
+import { Config, ConfigSystem } from '../config';
+
+export default async function api(fastify: FastifyInstance) {
+  fastify.get('/api', (req: FastifyRequest, reply: FastifyReply) => {
+    reply.status(200).send({
+      name: 'UMT API',
+      version: '1.0',
+      status: 'healthy'
+    });
+  });
+
+  fastify.get('/api/config', async (req: FastifyRequest, reply: FastifyReply) => {
+    try {
+      const { config } = ConfigFactory.getConfigFactory().loadConfig();
+      reply.status(200);
+      return config;
+    } catch (error) {
+      reply.status(404).send({ message: 'No configuration found' });
+    }
+
+  });
+
+  fastify.post('/api/config', async (req: FastifyRequest, reply: FastifyReply) => {
```

Comment:
I was able to post a config without any domains, which crashed the app after a logout. I was unable to post a new config after the crash happened

Reply:
resolved

Reply:
gj finding this case but there will be others. i'd bet on it.
this points to a deeper problem - how do you recover in production if an invalid config crashes the app? just another reason this pattern is concerning